terraform/deployments/govuk-publishing-platform/signon_bearer_tokens.tf

```diff
@@ -15,6 +15,180 @@ resource "random_password" "signon_admin_password" {
   special = true
 }
 
+#
+# SecretsManager Rotation Lambda
+#
+
+locals {
+  bearer_token_lambda_name = "bearer_token_rotater-${local.workspace}"
+}
+
+data "archive_file" "bearer_token_rotater" {
+  type        = "zip"
+  source_file = "${path.module}/../../../lambdas/signon_bearer_token_rotater.rb"
+  output_path = "${path.module}/../../../lambdas/signon_bearer_token_rotater.zip"
+}
+
+resource "aws_lambda_function" "bearer_token" {
+  function_name = local.bearer_token_lambda_name
+  role          = aws_iam_role.lambda_execution_role.arn
+  handler       = "${local.bearer_token_lambda_name}.handler"
+
+  filename         = data.archive_file.bearer_token_rotater.output_path
+  source_code_hash = data.archive_file.bearer_token_rotater.output_base64sha256
+
+  runtime = "ruby2.7"
+
+  vpc_config {
+    subnet_ids         = local.private_subnets
+    security_group_ids = [aws_security_group.signon_lambda.id]
+  }
+
+  environment {
+    variables = {
+      ADMIN_PASSWORD_KEY  = aws_secretsmanager_secret.signon_admin_password.arn
+      DEPLOY_EVENT_BUCKET = aws_s3_bucket.deploy_event_bucket.id
+      # TODO: Should be HTTPS
+      SIGNON_API_URL = "http://${module.signon.virtual_service_name}/api/v1"
+    }
+  }
+
+  # As recommended in Terraform docs
+  depends_on = [
+    aws_iam_role_policy_attachment.lambda_logs,
+    aws_iam_role_policy_attachment.vpc,
+    aws_cloudwatch_log_group.bearer_token,
+  ]
+
+  tags = merge(
+    local.additional_tags,
+    {
+      Name = "${local.bearer_token_lambda_name}-${var.govuk_environment}-${local.workspace}"
+    },
+  )
+}
+
+resource "aws_cloudwatch_log_group" "bearer_token" {
+  name              = "/aws/lambda/${local.bearer_token_lambda_name}"
+  retention_in_days = 90
+}
+
+# Adds a trust policy to the lambda function.
+resource "aws_lambda_permission" "allow_secretsmanager" {
+  statement_id  = "AllowExecutionFromSecretsManager"
+  action        = "lambda:InvokeFunction"
+  function_name = aws_lambda_function.bearer_token.function_name
+  principal     = "secretsmanager.amazonaws.com"
+  # depends_on    = [aws_secretsmanager_secret.bearer_token]
```

Comment:
Usually considered a bit iffy to merge commented-out code. Best to just remove it if it's not needed, or if it is gonna be needed later then add a TODO to explain why and what should happen in order for it to be uncommented.

Reply:
Thanks, I missed that!

Reply:
No need, I appreciated it :)